assets/js/components/notifications/BannerNotification/index.js

```diff
@@ -123,6 +124,10 @@ function BannerNotification( {
 			// Set the dismissed flag in cache without immediately hiding it.
 			await persistDismissal();
 		}
+
+		if ( onView && typeof onView === 'function' ) {
```

Comment:
We don't need to check the type of `onView` here as it will be checked by the `PropType`. We don't do it for the `onDismiss` so I think we can be consistent here.